### ISSUE TITLE
Change `.t` to `.elem_type`

### ISSUE THREAD

**Issue by Calvin-L** [CozySynthesizer/cozy#45]
The objects that represent collection types (`TBag`, `TSet`, and `TList`) use the attribute `t` for the type of their elements. This is ugly and is a bad name.

We should refactor the code to use `elem_type` instead. This will be a wide-reaching change and we should be careful we got it right.

**Comment by Calvin-L** [comment]
Yes, those should probably also get cleaner names.

(Also, I am not sure `TVector` is used anymore.)

**Comment by anhnamtran** [comment]
I'll take a look at `TVector` and do some cleaning up.

**Comment by anhnamtran** [comment]
Similarly, are most of the `.f`s `.fields` as well? I was looking through it and it seems to make sense. Since they're in `EArgMin` and so on...

**Comment by anhnamtran** [comment]
When I change `EArg(Min|Max).f` to `.key_function`, `EMap` and `EFlatMap` breaks. For now I will temporarily change their `.f` to `.key_function` as well.

**Comment by Calvin-L** [comment]
Yep, those `isinstance` checks are hacky. They exploit the fact that all of those AST nodes have `.e` and `.f` properties. You should fix them by duplicating those checks and their bodies for each different case.

I agree that, too, feels wrong. We should think about a better design for the places that happens.

**Comment by anhnamtran** [comment]
Everything should be covered in #47.
